src/elements/Text.tsx

```diff
@@ -0,0 +1,38 @@
+import type { TextNode } from '@prezly/slate-types';
+import React, { ReactNode } from 'react';
+
+const LINE_BREAKS = /\r\n|\r|\n/;
+
+export function Text({ bold, italic, subscript, superscript, text, underlined }: TextNode) {
+    let children: ReactNode = Text.preserveSoftBreaks(text);
+
+    if (bold) {
+        children = <strong>{children}</strong>;
+    }
+
+    if (italic) {
+        children = <em>{children}</em>;
+    }
+
+    if (subscript) {
+        children = <sub>{children}</sub>;
+    }
+
+    if (superscript) {
+        children = <sup>{children}</sup>;
+    }
+
+    if (underlined) {
+        children = <u>{children}</u>;
+    }
+
+    return <>{children}</>;
+}
+
+Text.preserveSoftBreaks = function (text: string): ReactNode {
+    const nodes = text.split(LINE_BREAKS).reduce<ReactNode[]>(function (result, part) {
```

Comment:
just a question, can somehow "pre" tag help, it also should preserve white spaces?

Reply:
@aspirisen `white-space: pre-wrap` can help. But I didn't want to rely on styles for this feature, as this is easy to miss in the consuming app.